test/teos/unit/test_internal_api.py

```diff
@@ -232,6 +248,139 @@ def test_get_appointment_non_existent(internal_api, stub):
+    uuid = uuid4().hex
+    internal_api.watcher.db_manager.store_responder_tracker(uuid, tracker.to_dict())
+
+    response = stub.get_all_appointments(Empty())
+    appointments = dict(response.appointments)
+
+    assert len(appointments.get("watcher_appointments")) == 0 and len(appointments.get("responder_trackers")) == 1
+    assert dict(appointments.get("responder_trackers")[uuid]) == tracker.to_dict()
+
+    # Delete the data
+    internal_api.watcher.db_manager.delete_responder_tracker(uuid)
+
+
+# FIXME: 194 will do with dummy appointments and trackers
+def test_get_all_appointments_both(clear_state, internal_api, generate_dummy_appointment, generate_dummy_tracker, stub):
+    # Data is pulled straight from the database, so we need to feed some
+    appointment, _ = generate_dummy_appointment()
+    uuid_appointment = uuid4().hex
+    internal_api.watcher.db_manager.store_watcher_appointment(uuid_appointment, appointment.to_dict())
+
+    tracker = generate_dummy_tracker()
+    uuid_tracker = uuid4().hex
+    internal_api.watcher.db_manager.store_responder_tracker(uuid_tracker, tracker.to_dict())
+
+    response = stub.get_all_appointments(Empty())
+    appointments = dict(response.appointments)
+
+    assert len(appointments.get("watcher_appointments")) == 1 and len(appointments.get("responder_trackers")) == 1
+    assert dict(appointments.get("watcher_appointments")[uuid_appointment]) == appointment.to_dict()
+    assert dict(appointments.get("responder_trackers")[uuid_tracker]) == tracker.to_dict()
+
+
+def test_get_tower_info_empty(clear_state, internal_api, stub):
+    response = stub.get_tower_info(Empty())
+    assert isinstance(response, GetTowerInfoResponse)
+    assert response.tower_id == Cryptographer.get_compressed_pk(internal_api.watcher.signing_key.public_key)
+    assert response.n_registered_users == 0
+    assert response.n_watcher_appointments == 0
+    assert response.n_responder_trackers == 0
+
+
+def test_get_tower_info(internal_api, stub, monkeypatch):
+    monkeypatch.setattr(internal_api.watcher.gatekeeper, "registered_users", {"uid1": {}})
+    monkeypatch.setattr(
+        internal_api.watcher,
+        "appointments",
+        {
+            "uid1": {"locator": "locator1", "user_id": "user_id1"},
+            "uid2": {"locator": "locator2", "user_id": "user_id2"},
+        },
+    )
+    monkeypatch.setattr(
+        internal_api.watcher.responder,
+        "trackers",
+        {
+            "uid1": {"penalty_txid": "txid1", "locator": "locator1", "user_id": "user_id1"},
+            "uid2": {"penalty_txid": "txid2", "locator": "locator2", "user_id": "user_id2"},
+            "uid3": {"penalty_txid": "txid3", "locator": "locator2", "user_id": "user_id3"},
+        },
+    )
+
+    response = stub.get_tower_info(Empty())
+    assert isinstance(response, GetTowerInfoResponse)
+    assert response.tower_id == Cryptographer.get_compressed_pk(internal_api.watcher.signing_key.public_key)
```

Comment:
There's a teos_pk defined globally in this test file.

Reply:
Done in 7e8482d915bd416d957abeda3a0d0e8746bc5e94.